modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/ParserServiceImpl.java

```diff
@@ -68,7 +66,7 @@ public ParsedResult parse(String query) {
         );
 
         if (shouldBeCached(queryType)) {
-            queryToParsedResultCache.put(query, result);
+            queryToParsedResultCache.computeIfAbsent(query, k -> result);
```

Comment:
Why do we use `compute` instead of `put`?

Reply:
I tried to make Cache interface minimal yet sufficient. But if it causes confusion, it's better to replace this call with `put`. 

Fixed.